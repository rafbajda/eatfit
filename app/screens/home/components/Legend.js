import React from 'react';
import emojis from '../../../shared/constants/emojis';
import {
    LegendDescription,
    LegendHeader,
    LegendIcon,
    LegendIconDescriptionHeader,
    LegendListElement,
    LegendRow
} from '../styles/legendStyles';
import I18n from 'i18n-js';

const Legend = () => {
    const {
        veryBadImpact,
        badImpact,
        neutralImpact,
        goodImpact,
        veryGoodImpact,
        t
    } = {
        ...emojis,
        ...I18n
    };
    const list = [
        {
            rightElement: (
                <LegendIconDescriptionHeader>
                    {t('info.veryGoodImpact')}
                </LegendIconDescriptionHeader>
            ),
            leftElement: <LegendIcon>{veryGoodImpact}</LegendIcon>
        },
        {
            rightElement: (
                <LegendIconDescriptionHeader>
                    {t('info.prettyGoodImpact')}
                </LegendIconDescriptionHeader>
            ),
            leftElement: <LegendIcon>{goodImpact}</LegendIcon>
        },

        {
            rightElement: (
                <LegendIconDescriptionHeader>
                    {t('info.ratherNeutralImpact')}
                </LegendIconDescriptionHeader>
            ),
            leftElement: <LegendIcon>{neutralImpact}</LegendIcon>
        },
        {
            rightElement: (
                <LegendIconDescriptionHeader>
                    {t('info.prettyBadImpact')}
                </LegendIconDescriptionHeader>
            ),
            leftElement: <LegendIcon>{badImpact}</LegendIcon>
        },
        {
            rightElement: (
                <LegendIconDescriptionHeader>
                    {t('info.veryBadImpact')}
                </LegendIconDescriptionHeader>
            ),
            leftElement: <LegendIcon>{veryBadImpact}</LegendIcon>
        }
    ];
    return (
        <LegendRow>
            <LegendHeader>{t('labels.instruction')}</LegendHeader>
            <LegendDescription>
                {t('info.instructionDetails')}
            </LegendDescription>
            {list.map((l, i) => (
                <LegendListElement
                    key={i}
                    leftElement={l.leftElement}
                    rightElement={l.rightElement}
                    bottomDivider
                    topDivider
                />
            ))}
        </LegendRow>
    );
};

export default Legend;
