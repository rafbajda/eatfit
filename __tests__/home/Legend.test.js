import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import Legend from '../../app/screens/home/components/Legend';

describe('Legend snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<Legend />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
