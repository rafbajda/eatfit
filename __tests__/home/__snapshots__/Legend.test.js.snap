// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend snapshot renders component 1`] = `
<View
  className="sc-bxivhb sc-gZMcBi hDnYZR"
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Text
    className="sc-gqjmRU iyQQPJ"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  >
    [missing "en.labels.instruction" translation]
  </Text>
  <Text
    className="sc-VigVT iRPIKY"
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
      }
    }
    uppercase={false}
  >
    [missing "en.info.instructionDetails" translation]
  </Text>
  <View
    className="sc-jzJRlG VQoua"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <Text
        className="sc-jTzLTM fXMUfF"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        😁
      </Text>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
      </View>
      <View
        width={16}
      />
      <Text
        className="sc-fjdhpX dunvmM"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        [missing "en.info.veryGoodImpact" translation]
      </Text>
    </View>
  </View>
  <View
    className="sc-jzJRlG VQoua"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <Text
        className="sc-jTzLTM fXMUfF"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        😊
      </Text>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
      </View>
      <View
        width={16}
      />
      <Text
        className="sc-fjdhpX dunvmM"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        [missing "en.info.prettyGoodImpact" translation]
      </Text>
    </View>
  </View>
  <View
    className="sc-jzJRlG VQoua"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <Text
        className="sc-jTzLTM fXMUfF"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        😐
      </Text>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
      </View>
      <View
        width={16}
      />
      <Text
        className="sc-fjdhpX dunvmM"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        [missing "en.info.ratherNeutralImpact" translation]
      </Text>
    </View>
  </View>
  <View
    className="sc-jzJRlG VQoua"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <Text
        className="sc-jTzLTM fXMUfF"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        😨
      </Text>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
      </View>
      <View
        width={16}
      />
      <Text
        className="sc-fjdhpX dunvmM"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        [missing "en.info.prettyBadImpact" translation]
      </Text>
    </View>
  </View>
  <View
    className="sc-jzJRlG VQoua"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <Text
        className="sc-jTzLTM fXMUfF"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        😱
      </Text>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
      </View>
      <View
        width={16}
      />
      <Text
        className="sc-fjdhpX dunvmM"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
      >
        [missing "en.info.veryBadImpact" translation]
      </Text>
    </View>
  </View>
</View>
`;
